### ISSUE TITLE
How to change the position of box in canvas

### ISSUE THREAD

**Issue by Super-Tree** [vispy/vispy#1411]
When I use
`box=vispy.scene.Box()`
to create a box in canvas, It seems that there is no parameters(class [vispy.scene.visuals.Box)](http://vispy.org/scene.html#module-vispy.scene.visuals) to specify the position of box,and I didn't find any vispy interfaces about it.How to edit the position of the box?

**Comment by kmuehlbauer** [comment]
Copying discussion from gitter for reference ([comment and ff.](https://gitter.im/vispy/vispy?at=5a4ee61a0163b02810b3c39a))
@Super-Tree 

AFAIK `BoxVisual` is a so-called `CompoundVisual`. It uses `MeshVisual` under the hood. In your case you would use `mesh = box1.mesh` to get the `MeshVisual` for the box-faces and `border = box1.border` for the box-edges. With that `MeshVisual` you can get the underlying `MeshData` object, which holds all vertices and alike (`meshdata = mesh.meshadata`. 

You then can manipulate the vertices (`vertices = meshdata.get_vertices`). This is not really straightforward, but might help you for the time being. 

I think BoxVisual needs some function to set the center and the orientation in space. @davidh-ssec @larsoner Any ideas from your side?

**Comment by kmuehlbauer** [comment]
Pinging @KelSolaar, who had the initial idea of box and plane and did the first implementation, and @bollu, who implemented the CompoundVisual changes.

Do you have some ideas where to implement position and orientation of BoxVisual in XYZ-space?

**Comment by Super-Tree** [comment]
`box1 = visual.sence.Box(width=1,height=2,depth=3, color=(0,0,1,0.3),edge_color='green')`
`mesh_box1 = box1.mesh.mesh_data`
`vertices = mesh_box1.get_vertices()`
`center=np.array([7.,2,-0.5],dtype=np.float32)`
`vtcs = np.add(vertices,center)`
`mesh_box1.set_vertices(vtcs)`

It works well,But the edges of the box seems not to change the origin position.
![vispy](https://user-images.githubusercontent.com/24618712/34600428-5d02d1ca-f232-11e7-9fba-0c5c76118db4.png)

**Comment by kmuehlbauer** [comment]
@Super-Tree Nice, you need to do the same thing with the `border`:

`mesh_border_box1 = box1.border.mesh_data`

**Comment by EtienneCmb** [comment]
Hey @Super-Tree and @kmuehlbauer 

Maybe the easiest way to do it is to use [STTransform](http://vispy.org/visuals.html?highlight=sttransform#vispy.visuals.transforms.STTransform) for scaling / translation or [MatrixTransform](http://vispy.org/visuals.html?highlight=sttransform#vispy.visuals.transforms.MatrixTransform) if you have to use rotations. Press +, -, (, ) to see the effect of the transformation.

```python
import numpy as np
from vispy import app, scene
from vispy.visuals import transforms

canvas = scene.SceneCanvas(keys='interactive', show=True)
vb = canvas.central_widget.add_view()
vb.camera = 'turntable'
vb.camera.rect = (-10, -10, 20, 20)

box = scene.visuals.Box(width=1, height=2, depth=3, color=(0, 0, 1, 0.3),
                        edge_color='green')
vb.add(box)

# Define a scale and translate transformation :
box.transform = transforms.STTransform(translate=(0., 0., 0.),
                                       scale=(1., 1., 1.))


@canvas.events.key_press.connect
def on_key_press(ev):
    tr = np.array(box.transform.translate)
    sc = np.array(box.transform.scale)
    if ev.text in '+':
        tr[0] += .1
    elif ev.text == '-':
        tr[0] -= .1
    elif ev.text == '(':
        sc[0] += .1
    elif ev.text == ')':
        sc[0] -= .1
    box.transform.translate = tr
    box.transform.scale = sc
    print('Translate (x, y, z): ', list(tr),
          '\nScale (x, y, z): ', list(sc), '\n')

if __name__ == '__main__':
    import sys
    if sys.flags.interactive != 1:
        app.run()
```

**Comment by kmuehlbauer** [comment]
@EtienneCmb Good work! Most elegant approach! I really haven't thought about about using `transform` (my bad).  Cool thing is, there is no code change needed at the `Visual`.

@Super-Tree That way you can hold the translation-vector (as well as scaling-  and rotation-vector) in your main program and set the `BoxVisual` at your convenience!

**Comment by EtienneCmb** [comment]
Transformations are amazing, just as [node](http://vispy.org/scene.html?highlight=node#module-vispy.scene.node). Define a node, set it as the parent of multiple visuals, set a transformation to this node which is then applied to all children... It's an awesome combination !

**Comment by kmuehlbauer** [comment]
Yes, I use that too, but my applications are somehow `static`. I do setup transformation and stuff only once and then it's fixed. No objects moving around, only `cameras`, but they work also behind the scenes.

@Super-Tree If this resolves your requirements, you may close this issue.

**Comment by Super-Tree** [comment]
Yes, I have tried your way and it works fine as well : )@EtienneCmb
Thanks for your help. @kmuehlbauer  @EtienneCmb

**Comment by rohitsaikiran91** [comment]
Hi, I tried Matrixtransform for rotation and STTransform for scaling the cube, but when the cube has an orientation and I try to scale, it becomes a rhombus. Any suggestions?

**Comment by djhoese** [comment]
@rohitsaikiran91 This issue is 1.5 years old. It would be easier to address your problem and help others in the future if you created a new issue and included a minimal code example for what you are trying to do and tell us what doesn't work. Thanks.
